libs-haskell/da-hs-base/src/DA/Service/Logger/Impl/GCP.hs

```diff
@@ -104,7 +123,8 @@ requestTimeout = 5_000_000
 -- Files used to record data that should persist over restarts.
 
 sentDataFile :: GCPState -> FilePath
-sentDataFile GCPState{gcpDamlDir} = gcpDamlDir </> ".sent_data"
+sentDataFile GCPState{..} =
```

Comment:
Definitely not something for this PR but I somewhat wonder if we might be better off in the long term just writing stuff to a `sqlite` database instead of messing around with a bunch of files pretending we can safely lock them.